solidity/test/bridge/Bridge.test.ts

```diff
@@ -4669,6 +4789,961 @@ describe("Bridge", () => {
+          pendingRedemptionsValue: 0,
+        })
+        await bridge
+          .connect(thirdParty)
+          .submitFraudChallenge(
+            fraudWalletPublicKey,
+            data.sighash,
+            data.signature.v,
+            data.signature.r,
+            data.signature.s,
+            {
+              value: fraudChallengeDepositAmount,
+            }
+          )
+      })
+
+      after(async () => {
+        await restoreSnapshot()
+      })
+
+      it("should revert", async () => {
+        await expect(
+          bridge
+            .connect(thirdParty)
+            .submitFraudChallenge(
+              fraudWalletPublicKey,
+              data.sighash,
+              data.signature.v,
+              data.signature.r,
+              data.signature.s,
+              {
+                value: fraudChallengeDepositAmount,
+              }
+            )
+        ).to.be.revertedWith("Fraud challenge already exists")
+      })
+    })
+
+    context(
+      "when the provided data is correct and amount of ether sent is enough",
+      () => {
+        let tx: Transaction
+
+        before(async () => {
+          await createSnapshot()
+          await bridge
+            .connect(governance)
+            .setFraudChallengeDepositAmount(fraudChallengeDepositAmount)
+          await bridge.setWallet(fraudWalletPublicKeyHash, {
+            state: 1,
+            pendingRedemptionsValue: 0,
+          })
+          tx = await bridge
+            .connect(thirdParty)
+            .submitFraudChallenge(
+              fraudWalletPublicKey,
+              data.sighash,
+              data.signature.v,
+              data.signature.r,
+              data.signature.s,
+              {
+                value: fraudChallengeDepositAmount,
+              }
+            )
+        })
+
+        after(async () => {
+          await restoreSnapshot()
+        })
+
+        it("should transfer ether from the caller to the bridge", async () => {
+          await expect(tx).to.changeEtherBalance(
+            thirdParty,
+            fraudChallengeDepositAmount.mul(-1)
+          )
+          await expect(tx).to.changeEtherBalance(
+            bridge,
+            fraudChallengeDepositAmount
+          )
+        })
+
+        it("should store the fraud challenge data", async () => {
+          const challengeKey = buildChallengeKey(
+            fraudWalletPublicKey,
+            data.sighash,
+            data.signature.v,
+            data.signature.r,
+            data.signature.s
+          )
+          const fraudChallenge = await bridge.fraudChallenges(challengeKey)
+          expect(fraudChallenge.challenger).to.equal(
+            await thirdParty.getAddress()
+          )
+          expect(fraudChallenge.depositAmount).to.equal(
+            fraudChallengeDepositAmount
+          )
+          expect(fraudChallenge.reportedAt).to.equal(await lastBlockTime())
+          expect(fraudChallenge.closed).to.equal(false)
```

Comment:
Let's divide sections using empty lines. We have three sections here: building key, fetching challenge, and assertions.

Reply:
Done in 61efd70c5ee84fcc4ac10154b2d2cd8c9af0a5aa.